docker/Dockerfile.spdk

```diff
@@ -0,0 +1,25 @@
+FROM fedora:36 AS base
+
+# Generic args
+ARG spdk_version
+ARG PROXY
+ARG NO_PROXY
+
+ENV http_proxy=$PROXY
+ENV https_proxy=$PROXY
+ENV no_proxy=$NO_PROXY
+
+RUN dnf install -y rpm-build librbd-devel git \
+    && dnf clean all
+
+COPY ./spdk /tmp/spdk
+
+RUN ls -l
+RUN ls -l /tmp
+
+COPY . . 
+
+RUN /docker/pre-install $spdk_version
+RUN /docker/post-install
+
+CMD ["/bin/bash"]
```

Comment:
@guptsanj I've been recently playing with the Dockerfile provided in the latest `spdk/spdk` repo and came up with something like the dockerfile below.

Lessons that I learnt:
* [Multi-stage dockerfile](https://docs.docker.com/build/building/multi-stage/) is optimal. It allows you to define multiple images based on others (hence reducing boilerplate/duplicate code) and benefits a lot of the internal caching that docker/podman provide.
* For the base layer I'd go for something long-term stable, like CentOS. I tested with CentOS Stream 8 (@oritwas tried with CentOS Stream 9, but no luck so far). Alternatively, @trociny and @idryomov suggested the existing `ceph` containers, which already come with the ceph packages installed.
* As discussed before: this Dockerfile (and the generated image) should be provided from within spdk project (and a Ceph Jenkins job or Github action will build that and publish the image to a quay.io repo). That said, in our current ceph/spdk branch, there are no Dockefiles, but in the `spdk/spdk:main` we have [some](https://github.com/spdk/spdk/blob/master/docker/build_base/Dockerfile) (that's only for tarball-based install, while my proposal below, as well as yours, work with an existing git codebase).
* As you did in the preinstall script, I'd also use the `rpm.sh` script (based on `rpmbuild`, I found is faster/leaner than the regular cmake-one, probably due to some testing skipped or different defaults...).
* For storing the generated artifacts (RPM packages), rather that copying things to/from `/tmp`, we can either reuse the RPMs from an existing layer (`spdk/spdk_rpm`), or export them outside (`DOCKER_BUILDKIT=1 docker build --target builder_local_rpm_exporter -o out/ .`).
The efficiently build the following Dockerfile, it's recommended to enable the `DOCKER_BUILDKIT` evolution:
`DOCKER_BUILDKIT=1 docker build -t <image_name> --target spdk_rpm .`

```Dockerfile
# syntax = docker/dockerfile:1.3
ARG BASE=quay.io/centos/centos:stream8
ARG BUILDER=rpm
ARG TYPE=local
ARG CACHE_DNF="/var/cache/dnf"
ARG CACHE_PIP="/root/.cache/pip"

FROM $BASE as base_local
WORKDIR /spdk
COPY . .


FROM base_local as builder_local_generic
RUN --mount=type=cache,target=$CACHE_DNF --mount=type=cache,target=$CACHE_PIP \
            scripts/pkgdep.sh
RUN ./configure
RUN make


FROM base_local as builder_local_rpm
RUN --mount=type=cache,target=$CACHE_DNF --mount=type=cache,target=$CACHE_PIP \
            yum install -y git python3 python3-devel rpm-build
RUN --mount=type=cache,target=$CACHE_DNF --mount=type=cache,target=$CACHE_PIP \
            rpmbuild/rpm.sh

FROM scratch as builder_local_rpm_exporter
COPY --from=builder_local_rpm /root/rpmbuild/rpm/ .

FROM builder_local_rpm as spdk_rpm
COPY --from=builder_local_rpm /root/rpmbuild/rpm/ /root/rpmbuild/rpm/
RUN yum install -y /root/rpmbuild/rpm/${HOSTTYPE}/*.rpm
```